scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -628,24 +630,66 @@ private Collection<ResourceType> getResourceTypes(final MissionModelId missionMo
   }
 
   /**
-   * serialize the given java object in a manner that can be used as a graphql argument value
-   *
-   * eg wraps strings or enums in quotes
-   *
-   * @param obj the object to serialize
+   * serialize the given string in a manner that can be used as a graphql argument value
+   * @param s the string to serialize
    * @return a serialization of the object suitable for use as a graphql value
    */
-  public String getGraphQLValueString(Object obj) {
+  public String serializeForGql(final String s) {
     //TODO: can probably leverage some serializers from aerie
-    if (obj instanceof String || obj instanceof Enum<?> || obj instanceof Instant) {
-      //TODO: (defensive) should escape contents of bare strings, eg internal quotes
-      //NB: Time::toString will format correctly as HH:MM:SS.sss, just need to quote it here
-      return "\"" + obj + "\"";
-    } else if (obj instanceof Duration dur) {
-      //NB: merlin uses durations in microseconds! (inconsistent with start_offset as a HH:MM:SS.sss string)
-      return Long.toString(dur.in(Duration.MICROSECOND));
-    } else {
-      return obj.toString();
-    }
+    //TODO: (defensive) should escape contents of bare strings, eg internal quotes
+    //NB: Time::toString will format correctly as HH:MM:SS.sss, just need to quote it here
+    return "\"" + s + "\"";
+  }
+
+  /**
+   * serialize the given duration in a manner that can be used as a graphql argument value
+   * @param d the duration to serialize
+   * @return a serialization of the object suitable for use as a graphql value
+   */
+  public String serializeForGql(final Duration d) {
+    //TODO: can probably leverage some serializers from aerie
+    //NB: merlin uses durations in microseconds! (inconsistent with start_offset as a HH:MM:SS.sss string)
+    return Long.toString(d.in(Duration.MICROSECOND));
+  }
+
+  public String serializeForGql(final SerializedValue value) {
```

Comment:
This method is only as strong as the methods it relies on (the 2 other `serializeForGql()` overloads above).
There's a to-do to correctly implement string escaping. This PR at least moves us away from using `serializedValue.toString()` to serialize a `SerializedValue` as a GQL-formatted value.

Reply:
@patkenneally @adrienmaillard I've created https://jira.jpl.nasa.gov/browse/AERIE-1949 to correctly implement string `"`/`'` escaping as well as GQL-injection sanitization.

Reply:
Edit: " " " as well as using GQL **variables** to avoid the need for sanitization in the first place.